### PR TITLE
Fix #886 issue

### LINKED ISSUE #886: Delete donor with recipient only if its the last one
## Project context 
There are two pages in this project: MATCHINGS and PATIENTS. 

PATIENTS page includes all donor and recipient pairs of people who are not compatible but that particular donor wanted to donate a kidney to his/her recipient, so they entered this program. 

MATCHINGS page is the page where we compute NEW donor -> recipient pairs that are now compatible. (Meaning we switch donors and recipients between each other, so recipients no longer have original donors)

## Problem
Currently, we allow for one recipient to have multiple original donors, but this can't be replicated through the UI, since we haven't yet done that. 

In case we are deleting a donor that belongs to a recipient that has multiple donors, we delete also all relevant donors and recipients with him.

The new behavior should be:

in case we delete the only donor of a recipient, delete also the recipient, otherwise delete only the donor

## How to replicate the problem
You can either create recipients with multiple donors through various endpoint calls or by populating the database with files from the `tests/resources/high_res_example_small_data_multiple_donors` folder.

### PR DESCRIPTION
**Pull request by jogobeny**

### CODE REVIEW

**Comment by kristinagalik** [comment]
hi @jogobeny :)

there was a small typo that i overlooked when editing the issue description - it should have been "in case we delete the only donor of a recipient, delete also the recipient, otherwise delete only the **donor**". i apologize.

File: txmatching/database/services/patient_service.py [hunk 1]
```diff
@@ -400,8 +400,10 @@ def delete_donor_recipient_pair(donor_id: int, txm_event_id: int):
     donor, maybe_recipient = get_donor_recipient_pair(donor_id, txm_event_id)
```
Comment (kristinagalik) [inline txmatching/database/services/patient_service.py:400]: here is an unused variable `donor`, either i would access the id of the donor through this variable, or get only the `maybe_recipient` object from the function :)

**Review by kristinagalik — CHANGES_REQUESTED** [review]
just a small comment, and also please add/fix tests and rebase :)

**Review by kristinagalik — COMMENTED** [review]
well done, the code is fine by me :) @kubantjan could you please review the changes too :)?

**Review by kubantjan — APPROVED** [review]

**Review by kristinagalik — APPROVED** [review]

**Pull request merged**